### PR TITLE
To remove swap_role()

### PR DESCRIPTION
**Pull request by JaeseungYeom**
To address the issue #207 
- change the only use of `swap_role()` to `set_role()`
- remove the `swap_role()` method
- change the `public` member methods for determining the number of samples to use to `protected` ones:
  `get_use_percent()`, `get_validation_percent()`, and `get_absolute_sample_count()`
However this depends on PR #208, which removes the external use of `get_use_percent()` under src/io/data_buffers. Those changes also show up here as I started new topic branch off from it.

### CODE REVIEW

File: include/lbann/data_readers/compound_data_reader.hpp [hunk 1]
```diff
@@ -92,12 +92,14 @@ class generic_compound_data_reader : public generic_data_reader {
...
   }
 
+/*
   void swap_role(std::string role) override {
     generic_data_reader::swap_role(role);
     for (auto&& reader : m_data_readers) {
       reader->swap_role(role);
     }
   }
+*/
```
Comment (ndryden) [inline include/lbann/data_readers/compound_data_reader.hpp]: Can you just delete this instead of commenting it out?

Comment (JaeseungYeom) [reply include/lbann/data_readers/compound_data_reader.hpp]: Will do

File: src/proto/proto_common.cpp [hunk 2]
```diff
@@ -2038,7 +2038,7 @@ void init_data_readers(bool master, const lbann_data::LbannPB& p, std::map<execu
         (*(ascii_reader *)reader_validation) = (*(ascii_reader *)reader);
       }
 
-      reader_validation->swap_role("validate");
+      reader_validation->set_role("validate");
```
Comment (bvanessen) [inline src/proto/proto_common.cpp]: I am not sure that what you are proposing here is correct.  The validation reader is a copy of the primary reader, but we then need to adjust the sample indices after the copy.

**Comment by bvanessen** [comment]
Why was this merged?

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470

**Comment by JaeseungYeom** [comment]
Sorry, it was unintentional. I was trying to merge the other PR on which this was dependent on. I can undo it.

File: src/proto/proto_common.cpp (hunk 2 above)
Comment (JaeseungYeom) [reply src/proto/proto_common.cpp]: By the way,  the adjustment of the sample indices is done by the following line reader_validation->use_unused_index_set() in a non-reversible way. What swap_role does is simply to swap m_use_percent and m_validation_percent, which is not a valid thing to do as these percents are not compatible.

**Pull request merged**